Editor/foleys_GUITreeEditor.cpp

```diff
@@ -27,28 +27,61 @@
  ==============================================================================
  */
 
-
 namespace foleys
 {
 
 GUITreeEditor::GUITreeEditor (MagicBuilder& builderToEdit)
-  : builder (builderToEdit)
+  : builder (builderToEdit), tree (builderToEdit.getGuiTree())
 {
-}
+    tree.addListener (this);
+    treeView.setRootItemVisible (false);
+    treeView.setMultiSelectEnabled (false);
 
-void GUITreeEditor::paint (juce::Graphics& g)
-{
-    g.setColour (juce::Colours::silver);
-    g.drawRect (getLocalBounds(), 1);
-    g.setColour (juce::Colours::white);
-    g.drawFittedText (TRANS ("GUI tree"), 2, 2, getWidth() - 4, 20, juce::Justification::centred, 1);
+    addNode.setEnabled (false);
+    addNode.onClick = [&]
+    {
+        // Show popup with types of nodes to add to the ValueTree: div, slider, button, etc
+        // add new node as a child of the currently selected node
+    };
```

Comment:
That's not how the nodes will come to live. We don't write an XML editor. There will be a toolbox, where you drag widgets into the actual GUI, and they will appear then in the tree...